comp/checks/windowseventlog/windowseventlogimpl/check/eventdatafilter/filter_test.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build windows
+
+package eventdatafilter
+
+import (
+	"fmt"
+
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestIsAllowedEventID(t *testing.T) {
+
+	f := &eventIDFilter{eventIDs: []int{1000, 2000}}
+	tcs := []struct {
+		eventID int
+		match   bool
+	}{
+		{1000, true},
+		{2000, true},
+		{3000, false},
+	}
+	for _, tc := range tcs {
+		match := f.isAllowedEventID(tc.eventID)
+		assert.Equal(t, tc.match, match)
+	}
+
+	f = &eventIDFilter{eventIDs: []int{}}
+	match := f.isAllowedEventID(1000)
+	assert.True(t, match)
+}
+
+func BenchmarkIsAllowedEventID(b *testing.B) {
+	sizeMatchSet := []int{10, 100, 1000, 10000}
+	for _, n := range sizeMatchSet {
+		b.Run(fmt.Sprintf("MatchEventID_%d", n), func(b *testing.B) {
+			// build a filter with n elements
+			f := &eventIDFilter{eventIDs: make([]int, n)}
+			for i := 0; i < n; i++ {
+				f.eventIDs[i] = i
+			}
+			b.ResetTimer()
+			for n := 0; n < b.N; n++ {
+				// set eventID to the size to imitate worst case scenario
+				f.isAllowedEventID(n)
+			}
+		})
+	}
+}
```

Comment:
This is probably most aplicable
BenchmarkIsAllowedEventID/MatchEventID_100                                                                                                                                       
BenchmarkIsAllowedEventID/MatchEventID_100-16           18950608                57.03 ns/op                                                                                      
BenchmarkIsAllowedEventID/MatchEventID_1000                                                                                                                                      
BenchmarkIsAllowedEventID/MatchEventID_1000-16           2913006               389.3 ns/op    

The numbers are not bad, but the question is if we can do better if and only if on some machine where we enable it and there is a high stream of such events continuously. And what if overall CPU overhead on high rate. Obviously, just enabling the feature alone will probably add higher overhead for system AND agent that filtering, but still would be interesting to know (and perhaps even document something).

Reply:
the primary bottleneck for all of the above is still `EvtFormatMessage`. See https://datadoghq.atlassian.net/browse/WINA-33

Reply:
> The numbers are not bad, but the question is if we can do better if and only if on some machine where we enable it and there is a high stream of such events continuously

I'm not sure what you mean. It's taking 389 nanoseconds to perform a (failing, worst case) match. That means we could match ~3,000,000 events per second. `EvtFormatMessage` bottlenecks us to ~3300 events/s, but even without `EvtFormatMessage` the best I've seen is ~66,000 events per second.

Reply:
Yeah, I understand. I did mean though a different overhead, invisible one. When we subscribe to high-rate source it alone will add a significant overhead of moving data from kernel or file (not sure) into Agent process implicitly. Sure, with humber posted for filtering it is visible and explicit overhead (part of that anyway) but by enabling the feature, all of a sudden we are making Agent to drink over a fire hydrant. We have no control over it though, but customer may not realize that it was their "fault" not hours.

To me overhead you a measure and throughput is only a part of the picture, another way. If customers start seeing Agent CPU spike, that what is that implicit overhead of drinking over firehose. Not sure what we can do about it beyond documentation. Or how bad it could be in some cases. In theory we can minimizing (maybe) this overhead by creating compound filtering (20-30 filters?) or even venture to ETW theretory. It is pure speculation at this point but it worth to be at least aware of that I think.